sentinel-ai-models-simple-openai/src/main/java/com/phonepe/sentinelai/models/SimpleOpenAIModel.java

```diff
@@ -213,6 +215,14 @@ yield runTools(message.getToolCalls(),
                             stats,
                             SentinelError.error(ErrorType.UNKNOWN_FINISH_REASON, response.getFinishReason()));
                 };
+
+                if(output == null && modelRunTerminationHandler.shouldTerminateEarly(modelSettings, context)) {
+                    log.debug("Early termination requested, terminating model run");
+                    return ModelOutput.error(
```

Comment:
The question is whether this is an error? Why is the val bound to output == null? Can you explain?